### PR TITLE
[FLINK-33796] Add ability to customize java version for python ci in connectors

### PR DESCRIPTION
**Pull request by snuyanzin**
Besides adding ability to customize java version ~~it also adds some tests for it~~ (currently without tests since python scripts expects `flink-python` folder and other acripts available)
The PR is done in a similar way it is done for non-python ci

### CODE REVIEW

**Comment by snuyanzin** [comment]
@gaborgsomogyi , @dianfu since you had a look at smilar PR can I ask you to hava a look here please?

File: .github/workflows/python_ci.yml [hunk 1]
```diff
@@ -23,6 +23,11 @@ on:
         description: "Flink version to test against."
         required: true
         type: string
+      jdk_version:
+        description: "Jdk version to test against."
+        required: false
+        default: 8, 11
```
Comment (pvary) [inline .github/workflows/python_ci.yml]: Does this add value to default to 2 different versions?
The python test infra creation takes serious time, and I am not sure that running the tests twice worth the additional resources committed.

Comment (snuyanzin) [reply .github/workflows/python_ci.yml]: yep, you're right
I changed to 8 only
thanks

**Comment by pvary** [comment]
@snuyanzin: Do we have a PR where this job has been tried out, and working? For testing my changes I started using the changes on my own repo to verify that everything is fine. https://github.com/pvary/flink-connector-kafka/actions

**Comment by snuyanzin** [comment]
@pvary yep, sure
here it is a link https://github.com/snuyanzin/flink-connector-kafka/actions/runs/7170817788

sorry I didn't put earlier

**Review by pvary — APPROVED** [review]
Thanks for the suggestion and the PR.
I still think we want to use it sparingly, but it is very good to have this option at hand

**Pull request merged**